### PR TITLE
 Updating INSTALL.md with coconut-svsm host setup, guest launch using…

### PR DESCRIPTION
**Pull request by ramagali24**
… direct boot and verify attestation & measurement using snpguest and igvmmeasure tools

### CODE REVIEW

File: scripts/launch_guest.sh [hunk 1]
```diff
@@ -8,10 +8,18 @@ set -e
 
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
+
+
```
Comment (stefano-garzarella) [inline scripts/launch_guest.sh:12]: Unrelated changes

File: scripts/launch_guest.sh [hunk 2]
```diff
@@ -8,10 +8,18 @@ set -e
...
 : "${IGVM:=$SCRIPT_DIR/../bin/coconut-qemu.igvm}"
+: "${KERNEL_BIN:="${guest_kernel}"}"
+: "${INITRD_BIN:="${GENERATED_INITRD_BIN}"}"
+
+GUEST_ROOT_LABEL="${GUEST_ROOT_LABEL:-cloudimg-rootfs}"
+GUEST_KERNEL_APPEND="root=LABEL=${GUEST_ROOT_LABEL} ro console=ttyS0"
+
 
-C_BIT_POS=`$SCRIPT_DIR/../utils/cbit`
+#C_BIT_POS=`$SCRIPT_DIR/../utils/cbit`
```
Comment (stefano-garzarella) [inline scripts/launch_guest.sh:22]: Why commenting this line?

File: scripts/launch_guest.sh [hunk 3]
```diff
@@ -8,10 +8,18 @@ set -e
 
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
+
+
 : "${QEMU:=qemu-system-x86_64}"
 : "${IGVM:=$SCRIPT_DIR/../bin/coconut-qemu.igvm}"
+: "${KERNEL_BIN:="${guest_kernel}"}"
+: "${INITRD_BIN:="${GENERATED_INITRD_BIN}"}"
```
Comment (stefano-garzarella) [inline scripts/launch_guest.sh:16]: I don't understand these definitions, usually this way is used to provide a default, but here you're putting the default to other environment variables not defined anywhere, what's the point?

File: scripts/launch_guest.sh [hunk 4]
```diff
@@ -115,6 +142,9 @@ $SUDO_CMD \
     $IMAGE_DISK \
     -nographic \
     -monitor none \
+    -kernel ${KERNEL_BIN} \
+    -initrd ${INITRD_BIN} \
+    -append "${GUEST_KERNEL_APPEND}"
```
Comment (stefano-garzarella) [inline scripts/launch_guest.sh:147]: By adding these parameters do we preserve the previous behavior of booting from disk?

File: Documentation/docs/installation/INSTALL.md [hunk 5]
```diff
@@ -436,6 +436,65 @@ of these limitations may be addressed in future updates.
 * Debugging is currently limited to the SVSM kernel itself. OVMF and the guest
   OS cannot be debugged using the SVSM GDB stub.
 
+Coconut-SVSM CI
+-------------
+
```
[the remark this answers is not in the document]
Comment (ramagali24) [reply Documentation/docs/installation/INSTALL.md:450]: Added description. Is that helpful for user before start proceeding further.

File: Documentation/docs/installation/INSTALL.md [hunk 6]
```diff
@@ -436,6 +436,65 @@ of these limitations may be addressed in future updates.
...
+Download the sev_utils script:
+```
+git clone https://github.com/ramagali24/sev-utils.git
+cd sev-utils/tools
+
+```
+
+To set up the host by building IGVM, SVSM, QEMU, OVMF, and the Linux kernel, use the following command:
+```
+./snp.sh --svsm setup-host
```
[the remark this answers is not in the document]
Comment (ramagali24) [reply Documentation/docs/installation/INSTALL.md:462]: As explained in the description, the setup-host script installs Coconut-SVSM-supported IGVM, OVMF, QEMU, the host and guest Linux kernels, and SVSM. This setup works for any Ubuntu workstation or a brand new Ubuntu host machine. It installs the necessary tools to build all these images and configures GRUB to ensure that the next reboot uses the Coconut-SVSM-supported host kernel. We perform all these steps to set up and launch the guest VM using the direct boot method. I am also working on other OS'es like Fedora, OpenSUSE support, once done will update.

File: scripts/launch_guest.sh (hunk 4 above)
Comment (ramagali24) [reply scripts/launch_guest.sh:145]: By specifying the kernel and initrd explicitly, you’re essentially bypassing the normal boot sequence that might read from a disk. It bypasses the bootloader (e.g., GRUB or LILO) that would typically be used to load the kernel from disk. Booting with these parameters, allows for testing with different kernel versions without altering the bootloader configuration on the virtual machine.

Comment (stefano-garzarella) [reply scripts/launch_guest.sh:143]: > By specifying the kernel and initrd explicitly, you’re essentially bypassing the normal boot sequence that might read from a disk. It bypasses the bootloader (e.g., GRUB or LILO) that would typically be used to load the kernel from disk. Booting with these parameters, allows for testing with different kernel versions without altering the bootloader configuration on the virtual machine.

Yep, I know that.

I try to rephrase my question. After this modification, can I use this script as I was using it before? ( i.e., by launching the kernel that is installed on my VM's disk).

We also sometimes launch QEMU without a disk, because we just want to test svsm, for example I just tried `make test-in-svsm` that uses that script and fails that way:
```
$ QEMU=/home/sgarzare/repos/snp-svsm-vtpm/qemu/build/qemu-system-x86_64 make test-in-svsm
cargo build  --target=x86_64-unknown-linux-gnu -p igvmbuilder
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.07s
cargo build  --target=x86_64-unknown-linux-gnu -p igvmmeasure
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.04s
./utils/gen_meta bin/meta.bin
cc    -c -o stage1/reset.o stage1/reset.S
cc -o bin/stage1-trampoline stage1/stage1-trampoline.o stage1/reset.o -nostdlib -Wl,--build-id=none -Wl,-Tstage1/stage1.lds -no-pie
objcopy -O binary bin/stage1-trampoline bin/stage1-trampoline.bin
LINK_TEST=1 cargo +nightly test  --no-default-features --features mstpm -p svsm --config 'target.x86_64-unknown-none.runner=["sh", "-c", "cp $0 ../target/x86_64-unknown-none/debug/svsm-test"]'
   Compiling svsm v0.1.0 (/home/sgarzare/repos/svsm-review/kernel)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 1.53s
     Running unittests src/lib.rs (target/x86_64-unknown-none/debug/deps/svsm-d7f23360a922e380)
objcopy -O elf64-x86-64 --strip-unneeded target/x86_64-unknown-none/debug/svsm-test bin/test-kernel.elf
cargo build --manifest-path kernel/Cargo.toml  --bin stage2
   Compiling svsm v0.1.0 (/home/sgarzare/repos/svsm-review/kernel)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.39s
objcopy -O binary "target/x86_64-unknown-none/debug/stage2" bin/stage2.bin
"target/x86_64-unknown-linux-gnu/debug/igvmbuilder" --sort --output bin/coconut-test-qemu.igvm --tdx-stage1 bin/stage1-trampoline.bin --stage2 bin/stage2.bin --kernel bin/test-kernel.elf qemu --snp --tdp
"target/x86_64-unknown-linux-gnu/debug/igvmmeasure" bin/coconut-test-qemu.igvm measure

===============================================================================================================
igvmmeasure 'bin/coconut-test-qemu.igvm'
Launch Digest: 5CF035524AF0DABE7F1C140393C866AA8FACAD8EA34F954C476ADACCB843684CCEE20A780283E8DDAF2E40E71F7E05E8
===============================================================================================================

cp -f "target/x86_64-unknown-linux-gnu/debug/igvmmeasure" bin/igvmmeasure
./scripts/test-in-svsm.sh
=============================
Launching SVSM guest
=============================
QEMU:         /home/sgarzare/repos/snp-svsm-vtpm/qemu/build/qemu-system-x86_64
QEMU Version: 8.2.0
IGVM:         /home/sgarzare/repos/svsm-review/scripts/../bin/coconut-test-qemu.igvm
IMAGE:        
KERNEL:       
INITRD:       
APPEND:       root=LABEL=cloudimg-rootfs ro console=ttyS0
=============================
Press Ctrl-] to interrupt
=============================
qemu: could not open kernel file '-initrd': No such file or directory
```

What I mean is that you need to add these new parameters to the QEMU command only when they are specified, and allow the script to be used as we used it before.

File: scripts/launch_guest.sh [hunk 7]
```diff
@@ -10,6 +10,11 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 : "${QEMU:=qemu-system-x86_64}"
 : "${IGVM:=$SCRIPT_DIR/../bin/coconut-qemu.igvm}"
+: "${KERNEL_BIN:="${guest_kernel}"}" ## Incase your work environment is ubuntu, guest_kernel is your ubuntu kernel image "vmlinuz-6.8.0-snp-guest-bc4de28e0cc1" o
+: "${INITRD_BIN:="${GENERATED_INITRD_BIN}"}" #Incase your work environment is ubuntu, GENERATED_INITRD_BIN is your generated guest linux "initrd.img-6.8.0-snp-guest-bc4de28e0cc1"
```
Comment (stefano-garzarella) [inline scripts/launch_guest.sh:14]: Again, I still don't understand the meaning of these definitions. Why do the default of these two new variables depend on other variables?

File: Documentation/docs/installation/INSTALL.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (stefano-garzarella) [reply Documentation/docs/installation/INSTALL.md:450]: Again, I don't think this is the right place for these steps.
These only work on a specific distro, whereas we tried to have all scripts, instructions, independent of distros. So IMO better a blog post or another file in the Documentation folder, for example specific to Ubuntu.

**Review by stefano-garzarella — COMMENTED** [review]
Please squash the fixes in the previous commit where you introduced the changes.
